packages/config/src/options.js

```diff
@@ -88,6 +88,26 @@ export function getNuxtConfig (_options) {
 
   defaultsDeep(options, nuxtConfig)
 
+  // Target
+  options.target = options.target || 'server'
+  if (!Object.values(TARGETS).includes(options.target)) {
+    consola.warn(`Unknown target: ${options.target}. Falling back to server`)
+    options.target = 'server'
+  }
+
+  // User server middleware warning if target=static
+  if (options.target === TARGETS.static && options.serverMiddleware.length) {
+    consola.warn(`serverMiddleware are not recommended in static target and can lead to side effects`)
```

Comment:
Suggestion for better wording: serverMiddleware does _not_ work with a static target.

Reply:
Also what if we automatically empty this array or disable all serverMiddleware at `@nuxt/server` level to prevent users going wrong?

Reply:
Well the point is to discourage user `serverMiddleware`,  but don't remove the modules `serverMiddleware` since they can work in devMode (pushing the `serverMiddleware` only in dev)

Reply:
What's the use case of server middleware for full-static mode? If the case is during production generate then IMO they should not be warned/discouraged. If not, why only supporting in dev mode?

Reply:
I thought the reason for introducing a new `static` target is giving near to production experience during development so developers don't do something wrong/unsupported.

Reply:
Indeed, it could be used for production generate as well, thinking of NuxtPress for instance.

Reply:
I will remove the warning and we will just have to add a note in the documentation of the serverMiddleware :)